plugin/src/main/java/org/opensearch/ml/action/prediction/TransportPredictionTaskAction.java

```diff
@@ -159,6 +165,44 @@ public void onFailure(Exception e) {
         }
     }
 
+    private void deployRemoteModel(
+        String modelId,
+        FunctionName functionName,
+        ActionListener<MLTaskResponse> wrappedListener,
+        MLPredictionTaskRequest mlPredictionTaskRequest
+    ) {
+        String[] workerNodes = mlModelManager.getWorkerNodes(modelId, functionName, true);
+        if (workerNodes != null && workerNodes.length != 0) {
+            return;
+        }
+
+        MLDeployModelRequest deployModelRequest = MLDeployModelRequest.builder().modelId(modelId).async(false).dispatchTask(true).build();
+        ActionListener<MLDeployModelResponse> deployModelActionListener = ActionListener.wrap(deployModelResponse -> {
+            // Deployment failed, existing
+            if (!deployModelResponse.getStatus().equals(MLTaskState.COMPLETED.name())) {
+                wrappedListener
+                    .onFailure(
+                        new IllegalArgumentException(
+                            "Model not ready yet. Please run this first: POST /_plugins/_ml/models/" + modelId + "/_deploy"
+                        )
+                    );
+                return;
+            }
+            // The DeployModel is async, set this maximum wait time for deployment to finish
+            long startTime = System.currentTimeMillis();
+            long maxDuration = 100; // 1 second in milliseconds
+            while (workerNodes == null || workerNodes.length == 0) {
```

Comment:
It looks like you're missing a sleep statement inside the while loop. Without it, the loop may consume a lot of CPU resources since it will continuously check the condition without any delay. Adding a Thread.sleep() statement inside the loop would be beneficial to reduce CPU usage.